core_info.c

```diff
@@ -873,6 +873,33 @@ void core_info_list_get_supported_cores(core_info_list_t *core_info_list,
    *num_infos = supported;
 }
 
+/*
+ * Matches core path A and B "base" filename (ignoring everything after _libretro)
+ *
+ * Ex:
+ *   snes9x_libretro.dll and snes9x_libretro_android.so are matched
+ *   snes9x__2005_libretro.dll and snes9x_libretro_android.so are NOT matched
+ */
+bool core_info_match_base_filename(const char* core_path_a, const char* core_path_b)
+{
+   if (!core_path_a || !core_path_b)
+      return false;
+
+   {
+      const char* core_path_basename_a = path_basename(core_path_a);
+      const char* core_path_basename_b = path_basename(core_path_b);
+      const char* libretro_pos = core_path_basename_a != NULL ? strstr(core_path_basename_b, "_libretro") : NULL;
+      if (libretro_pos)
+      {
+         size_t core_base_filename_length = libretro_pos - core_path_basename_a;
+         if (strncmp(core_path_basename_a, core_path_basename_b, core_base_filename_length) == 0)
```

Comment:
We have a function for this string comparison: `static INLINE bool string_starts_with_size(const char *str, const char *prefix, size_t size)`

Reply:
Done!